packages/react-menu/src/components/Menu/Menu.types.ts

```diff
@@ -119,7 +119,12 @@ export interface MenuState extends MenuProps {
 /**
  * Data attached to open/close events
  */
-export interface OnOpenChangeData extends Pick<MenuState, 'open'> {}
+export interface OnOpenChangeData extends Pick<MenuState, 'open'> {
+  /**
+   * Indicates whether the change of state was a keyboard interaction
+   */
+  keyboard: boolean;
```

Comment:
Isn't it enough checking if the provided event is a Keyboard event?

Reply:
no because `Enter` and `Spacebar` trigger click events, just like native

Reply:
It will actually be the same behaviour, if 1st rule of aria is followed and the default slot for `MenuItem` becomes `button`